crates/ra_hir/src/code_model.rs

```diff
@@ -773,13 +773,17 @@ impl Const {
 
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct ConstData {
-    pub(crate) name: Name,
+    pub(crate) name: Option<Name>,
     pub(crate) type_ref: TypeRef,
 }
 
 impl ConstData {
-    pub fn name(&self) -> &Name {
-        &self.name
+    pub fn name(&self) -> Option<&Name> {
+        self.name.as_ref()
+    }
+
+    pub fn is_unnamed(&self) -> bool {
```

Comment:
Let's remote this function, I think `name().is_none()` is a better name for this than `is_unnamed` :)